src/js/components/Calendar/Calendar.js

```diff
@@ -175,28 +175,31 @@ const Calendar = forwardRef(
               direction: 'up',
               weeks: daysApart(targetDisplayBounds[1], displayBounds[1]) / 7,
             });
-            animating = true;
+            setAnimating(true);
           }
         }
-
-        if (animating) {
-          // Wait for animation to finish before cleaning up.
-          const timer = setTimeout(
-            () => {
-              setDisplayBounds(targetDisplayBounds);
-              setTargetDisplayBounds(undefined);
-              setSlide(undefined);
-            },
-            400, // Empirically determined.
-          );
-          return () => clearTimeout(timer);
-        }
         return undefined;
       }
 
       setSlide(undefined);
       return undefined;
-    }, [displayBounds, targetDisplayBounds]);
+    }, [animating, displayBounds, targetDisplayBounds]);
+
+    useEffect(() => {
```

Comment:
Few words of why this useEffect is needed and what it is doing will be helpful.

Reply:
Added comment to accompany the useEffect.